packages/amplication-server/src/core/billing/billing.service.spec.ts

```diff
@@ -75,8 +75,103 @@ describe("BillingService", () => {
     );
   });
 
+  it("should throw exceptions on number of projects if the workspace has no entitlement to bypass code generation limitation and the current project is out of the limitation based on the number of allowed projects and its creation date", async () => {
+    const workspaceId = "id";
+    const projectsPerWorkspaceLimit = 1;
+
+    const spyOnServiceGetBooleanEntitlement = jest
+      .spyOn(service, "getBooleanEntitlement")
+      .mockResolvedValue({
+        hasAccess: false,
+      } as BooleanEntitlement);
+
+    const spyOnServiceGetMeteredEntitlement = jest
+      .spyOn(service, "getMeteredEntitlement")
+      .mockResolvedValue({
+        hasAccess: false,
+        usageLimit: projectsPerWorkspaceLimit,
+      } as MeteredEntitlement);
+
+    const user: User = {
+      id: "user-id",
+      account: {
+        id: "account-id",
+        createdAt: new Date(),
+        updatedAt: new Date(),
+        email: "email",
+        firstName: "first-name",
+        lastName: "last-name",
+        password: "password",
+      },
+      createdAt: new Date(),
+      updatedAt: new Date(),
+      isOwner: true,
+    };
+
+    const projects: Project[] = [
+      {
+        id: "project-id-1",
+        name: "project-1",
+        workspaceId: workspaceId,
+        useDemoRepo: false,
+        createdAt: new Date("2023-01-01"),
+        updatedAt: new Date(),
+      },
+      {
+        id: "project-id-2",
+        name: "project-2",
+        workspaceId: workspaceId,
+        useDemoRepo: false,
+        createdAt: new Date("2023-01-02"),
+        updatedAt: new Date(),
+      },
+    ];
+
+    await expect(
+      service.validateSubscriptionPlanLimitationsForWorkspace(
+        workspaceId,
+        user,
+        "project-id-2",
+        projects
+      )
+    ).rejects.toThrow(
+      new ValidationError("LimitationError: Allowed projects per workspace: 1")
+    );
+
+    expect(spyOnServiceGetBooleanEntitlement).toHaveBeenCalledTimes(1);
+    expect(spyOnServiceGetBooleanEntitlement).toHaveBeenCalledWith(
+      workspaceId,
+      BillingFeature.IgnoreValidationCodeGeneration
+    );
+    await expect(
```

Comment:
This expect is not needed since it's testing the test (or the mocking framework)